### PR TITLE
Canbomb suicide fix

### PR DESCRIPTION
**Pull request by Kaan-NN**
[BUG] [PLAYER-ACTIONS]
## About the PR
Fixes primed canbombs with atmos analyzer leaking air when suiciding but not exploding

## Why's this needed?
Bug fix for #11629

### CODE REVIEW

**Comment by pali6** [comment]
Yeah no. I don't feel like making it possible to immediately detonate a canbomb and skip the countdown is a good idea.

**Comment by BatElite** [comment]
Although it sounds like this isn't going to make, this seems too obscure to warrant a changelog

**Comment by Kaan-NN** [comment]
> Yeah no. I don't feel like making it possible to immediately detonate a canbomb and skip the countdown is a good idea.

It is already possible to do so by just opening the valve manually but I could also just remove that entirely if that would be preferred

File: code/modules/atmospherics/portable/canister.dm [hunk 1]
```diff
@@ -36,15 +36,25 @@
 
 	custom_suicide = 1
 	suicide(var/mob/user as mob)
+		if(src.destroyed)
+			return
 		if (!src.user_can_suicide(user))
 			return FALSE
+		usr.visible_message("<span class='alert'><b>[user] attempts to reach the valve with [his_or_her(user)] mouth to release some pressure!</b></span>")
```
Comment (pali6) [inline code/modules/atmospherics/portable/canister.dm:43]:
```suggestion
		user.visible_message("<span class='alert'><b>[user] attempts to reach the valve with [his_or_her(user)] mouth to release some pressure!</b></span>")
```

File: code/modules/atmospherics/portable/canister.dm [hunk 2]
```diff
@@ -36,15 +36,25 @@
...
+		if (src.det)
+			if (!src.det.part_fs.timing || src.det.defused)
+				boutput(usr, "<span class='alert'>You try to reach the valve with your mouth but the failsafe prevents you from reaching it.<br><i>Looks like priming the bomb might make it accessible to you...?</i></span>")
```
Comment (pali6) [inline code/modules/atmospherics/portable/canister.dm:46]:
```suggestion
				boutput(user, "<span class='alert'>You try to reach the valve with your mouth but the failsafe prevents you from reaching it.<br><i>Looks like priming the bomb might make it accessible to you...?</i></span>")
```

File: code/modules/atmospherics/portable/canister.dm [hunk 3]
```diff
@@ -36,15 +36,25 @@
...
+			return
 		if (src.release_pressure < 5*ONE_ATMOSPHERE || MIXTURE_PRESSURE(src.air_contents) < 5*ONE_ATMOSPHERE)
 			boutput(user, "<span class='alert'>You hold your mouth to the release valve and open it. Nothing happens. You close the valve in shame.<br><i>Maybe if you used more pressure...?</i></span>")
-			return FALSE
-		user.visible_message("<span class='alert'><b>[user] holds [his_or_her(user)] mouth to [src]'s release valve and briefly opens it!</b></span>")
+			return
 		src.valve_open = TRUE
-		user.gib()
-		return TRUE
+		usr.gib()
```
Comment (pali6) [inline code/modules/atmospherics/portable/canister.dm:57]:
```suggestion
		user.gib()
```

**Review by pali6 — APPROVED** [review]

**Pull request merged**